corehq/apps/style/static/style/js/hq_extensions.jquery.js

```diff
@@ -44,9 +44,43 @@
     $.fn.closest_form = function () {
         return this.closest('form, .form');
     };
+
     $.fn.my_serialize = function () {
         var data = this.find('[name]').serialize();
         return data;
     };
 
+    $.fn.addSpinnerToButton = function () {
+        $(this).prepend('<i class="fa fa-refresh fa-spin icon-refresh icon-spin"></i> ');
+    };
+
+    $.fn.removeSpinnerFromButton = function () {
+        $(this).find('i').remove();
+    };
+
+    $.fn.disableButtonNoSpinner = function () {
+        $(this).attr('disabled', 'disabled')
+            .addClass('disabled');
+    };
+
+    $.fn.disableButton = function () {
+        $(this).disableButtonNoSpinner();
+        $(this).addSpinnerToButton();
+    };
+
+    $.fn.enableButton = function () {
+        $(this).removeSpinnerFromButton();
+        $(this).removeClass('disabled')
+            .removeAttr('disabled');
+    };
+
+    $.fn.koApplyBindings = function (context) {
```

Comment:
There isn't a simple answer to this, and it depends on what we do with concatenating files, but it's worth remembering that a cached monolith can be better than a non-cached smaller file.